shell/platform/darwin/ios/framework/Source/platform_message_response_darwin.mm

```diff
@@ -4,6 +4,47 @@
 
 #import "flutter/shell/platform/darwin/ios/framework/Source/platform_message_response_darwin.h"
 
+/// A proxy object that behaves like NSData represented in a Mapping.
+/// This isn't a subclass of NSData because NSData is in a class cluster.
+@interface FlutterMappingData : NSObject
+@end
+
+@implementation FlutterMappingData {
+  NSData* _data;
+  std::unique_ptr<fml::Mapping> _mapping;
+}
+
++ (NSData*)dataWithMapping:(std::unique_ptr<fml::Mapping>)mapping {
+  return (NSData*)[[[FlutterMappingData alloc] initWithMapping:std::move(mapping)] autorelease];
+}
+
+- (instancetype)initWithMapping:(std::unique_ptr<fml::Mapping>)mapping {
+  self = [super init];
+  if (self) {
+    const void* rawData = mapping->GetMapping();
+
+    // Const cast is required because the NSData API requires it despite
+    // guarentees that the buffer won't be deleted or modified.
+    _data = [[NSData alloc] initWithBytesNoCopy:const_cast<void*>(rawData)
+                                         length:mapping->GetSize()
+                                   freeWhenDone:NO];
+
+    _mapping = std::move(mapping);
+  }
+  return self;
+}
+
+- (void)dealloc {
+  [_data release];
+  [super dealloc];
+}
+
+- (id)forwardingTargetForSelector:(SEL)aSelector {
+  return _data;
+}
+
```

Comment:
The respondsToSelector+forwardInvocation is a different slower forwarding mechanism.

Yea, if the code ever did reflection on the NSData it could be a problem.  Subclassing class clusters isn't clear, they really push forwarding in the documentation.

Reply:
Correctness is more important that speed though, and I find it *very* unlikely that the cost of calls to this NSData object is going to even more than a tiny, tiny fraction of the end-to-end performance of platform channels.

Reply:
I was mistaken, because NSData is a class cluster this isn't actually a problem.  Reflection doesn't work on proper NSData classes anyways.  If you do the following:
```objc
#include <Foundation/Foundation.h>

int main() {
  NSData* data = [[NSData alloc] init];
  NSLog(@"%@", data.class);
}
```
it prints out `_NSZeroData`.  You can't do reflection on NSData so this is as correct as can be.

That said, I'm still going to investigate the alternative we discussed since it is cleaner.

Reply:
> You can't do reflection on NSData

You've demonstrated one specific call. What about `isKindOfClass:`? And much more importantly, how does your implementation handle `respondsToSelector`?

Reply:
Yea, I'd have to override `respondsToSelector`.  To get it to respond to `isKindOfClass:` I'd have to either override all methods, or fiddle with the `objc_class`.  I put a pin in this while I research the broader question: https://docs.google.com/document/d/1oNLxJr_ZqjENVhF94-PqxsGPx0qGXx-pRJxXL6LSagc/edit